File: app/src/main/java/com/bilulo/androidtest04/ui/login/router/LoginRouter.java
package com.bilulo.androidtest04.ui.login.router;

import android.content.Intent;

import com.bilulo.androidtest04.data.model.UserAccountModel;
import com.bilulo.androidtest04.ui.list.view.ListActivity;
import com.bilulo.androidtest04.ui.login.contract.LoginContract;
import com.bilulo.androidtest04.ui.login.view.LoginActivity;
import com.bilulo.androidtest04.utils.SharedPreferencesUtil;

import java.lang.ref.WeakReference;

public class LoginRouter implements LoginContract.RouterContract {
    public WeakReference<LoginActivity> activity;

    private void startListActivity(UserAccountModel userAccountModel) {
        activity.get().startActivity(getListActivityIntent(userAccountModel));
        activity.get().finish();
    }

    public Intent getListActivityIntent(UserAccountModel userAccountModel) {
        Intent intent = new Intent(activity.get(), ListActivity.class);
        intent.putExtra(ListActivity.EXTRA_USER_ACCOUNT, userAccountModel);
        return intent;
    }

    public void saveUsernameInSharedPreferences(String username) {
        SharedPreferencesUtil.saveString(activity.get(), LoginActivity.KEY_USER, username);
    }

    @Override
    public void loginSuccessful(UserAccountModel userAccountModel, String username) {
        saveUsernameInSharedPreferences(username);
        startListActivity(userAccountModel);
    }
}

File: app/src/main/java/com/accenture/primo/bankapp/ui/login/LoginRouter.kt
package com.accenture.primo.bankapp.ui.login

import android.app.Activity
import android.content.Intent
import com.accenture.primo.bankapp.model.User

import com.accenture.primo.bankapp.ui.login.contracts.ILoginRouter
import com.accenture.primo.bankapp.ui.main.MainActivity
import com.accenture.primo.bankapp.EXTRA_KEY_LOGIN

class LoginRouter(private val activity: Activity): ILoginRouter {
    companion object {
        private const val LOGIN_EXTRA_KEY = EXTRA_KEY_LOGIN
    }

    override fun showNextScreen(user: User){
        var intent : Intent = Intent(activity, MainActivity::class.java)
        intent.putExtra(LOGIN_EXTRA_KEY, user)
        activity.startActivity(intent)
        //intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        //intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
        activity.finish()
    }

}
